clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -5996,6 +5996,12 @@ void Sema::AddIntelFPGAMaxReplicatesAttr(Decl *D, const AttributeCommonInfo &CI,
     // attributes are incompatible.
     if (checkAttrMutualExclusion<IntelFPGARegisterAttr>(*this, D, CI))
       return;
+
+    // If the declaration does not have an [[intel::fpga_memory]]
+    // attribute, this creates one as an implicit attribute.
+    if (!D->hasAttr<IntelFPGAMemoryAttr>())
+      D->addAttr(IntelFPGAMemoryAttr::CreateImplicit(
+          Context, IntelFPGAMemoryAttr::Default));
```

Comment:
Does it make sense to do this outside the if condition? i.e. we want to do this regardless of E's value dependency, no?

Reply:
Good catch! Same goes for checking for mutual exclusion, I suppose.

Reply:
yes, I will fix this and upload new patch.

Reply:
Done